feat(linkbdd): implement bdd for link

TheoPascoli opened the pull request:
**To discuss** 

The hybrid dao is useful to perform the migration from files to bdd step by step and to have some models stored in bdd while the others are still stored in the files.

The migration is perform "on the fly" while using the app. If you import a study, all objects with bdd implementation are saved in bdd. 
When exporting a study, everything that is in bdd is saved in the files, then zipped.

When using an old study, when you try to get a link, it wont be found in bdd, so it will fetch in files, then save them in bdd for the next times. So you will fetch old models in file only one time per old study. 

In the end, when everything is migrated, the hybrid dao is renamed to bdddao for example.

Note : some tests has been modified poorly, just to make them pass ... To change later

TheoPascoli committed:
build(sqlalchemy): remove unused ignore comment

TheoPascoli committed:
build(sqlalchemy): change study type for rawstudy and variantstudy

TheoPascoli committed:
build(sqlalchemy): finalise last mypy issues

TheoPascoli committed:
Merge branch 'dev' into build/complete-bump-to-sqlalchemy-2.0

TheoPascoli committed:
build(sqlalchemy): ruff check

TheoPascoli committed:
build(sqlalchemy): query fix

TheoPascoli committed:
Merge branch 'dev' into build/complete-bump-to-sqlalchemy-2.0

TheoPascoli committed:
build(sqlalchemy): more query fix

TheoPascoli committed:
build(sqlalchemy): more and more query fix

TheoPascoli committed:
build(sqlalchemy): more more and more query fix

TheoPascoli committed:
Merge remote-tracking branch 'origin/dev' into build/complete-bump-to-sqlalchemy-2.0

TheoPascoli committed:
build(sqlalchemy): more and more and more and more query fix

TheoPascoli committed:
build(sqlalchemy): more and more and more and more and more query fix

TheoPascoli committed:
build(sqlalchemy): comment ça mon ruff

TheoPascoli committed:
Merge branch 'dev' into build/complete-bump-to-sqlalchemy-2.0

TheoPascoli committed:
Merge branch 'dev' into build/complete-bump-to-sqlalchemy-2.0

TheoPascoli committed:
build(sqlalchemy): fix feedback

TheoPascoli committed:
Merge remote-tracking branch 'origin/dev' into build/complete-bump-to-sqlalchemy-2.0

# Conflicts:
#	antarest/matrixstore/model.py

TheoPascoli committed:
build(sqlalchemy): fix security warning

TheoPascoli committed:
build(sqlalchemy): fix security warning

TheoPascoli committed:
Merge branch 'dev' into build/complete-bump-to-sqlalchemy-2.0

TheoPascoli committed:
Merge branch 'dev' into build/complete-bump-to-sqlalchemy-2.0

TheoPascoli committed:
Merge branch 'dev' into build/complete-bump-to-sqlalchemy-2.0

TheoPascoli committed:
Merge branch 'dev' into build/complete-bump-to-sqlalchemy-2.0

sylvlecl committed:
fix(matrix-gc): fix singleton services creation

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

TheoPascoli committed:
Merge remote-tracking branch 'origin/dev' into dev

TheoPascoli committed:
Merge branch 'dev' into build/complete-bump-to-sqlalchemy-2.0

TheoPascoli committed:
build(sqlalchemy): fix cast issues

TheoPascoli committed:
Merge branch 'dev' into build/complete-bump-to-sqlalchemy-2.0

TheoPascoli committed:
Merge remote-tracking branch 'origin/dev' into dev

TheoPascoli committed:
feat(link-bdd): add link bdd implementation

TheoPascoli committed:
feat(link-bdd): wip

TheoPascoli committed:
build(sqlalchemy): fix issues

TheoPascoli committed:
feat(link-bdd): wip

sylvlecl committed:
fix(tests): fix random failure in matrix gc tests

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
fix(test): fix unit test

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

TheoPascoli committed:
Merge remote-tracking branch 'origin/fix/fix-random-failure-matrix-gc-tests' into build/complete-bump-to-sqlalchemy-2.0

TheoPascoli committed:
Merge branch 'build/complete-bump-to-sqlalchemy-2.0' into feat/implement-bdd-for-link

TheoPascoli committed:
Merge remote-tracking branch 'origin/dev' into feat/implement-bdd-for-link

TheoPascoli committed:
feat(linkbdd): implement bdd support for links

TheoPascoli committed:
Merge branch 'dev' into feat/implement-bdd-for-link

skamril committed:
fix(ui-sts): preserve active tab when saving a matrix (#2674)

sylvlecl committed:
chore(study)!: remove unused validation feature (#2670)

Removes dead code which aimed at validating the content of a file study.

Reasons are:
 - it is not used and not maintained (lacks tests)
 - it is often poorly implemented (checking only types)
 - it duplicates a lot of validation implemented at the pydantic model
level
 - it should be re-implemented at the DAO / import level
 - it's quite a lot of code :)

---------

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

skamril committed:
fix(ui): preserve tab state by passing ReactNode to TabsView content (#2676)

using `content: React.FunctionComponent` causes tab content to be recreated on each render, resulting in unnecessary unmounting and remounting of tab components. This leads to loss of internal state and a less consistent user experience when switching tabs or updating items.

skamril committed:
fix(ui-optimization): add new options for 'exportMps' field (#2669)

options available since v8.3

ANT-3569

mehdiwahada committed:
feat(matrix): API to retrieve all matrix references (#2673)

TheoPascoli committed:
feat(solver): add new solver options for linear and quadratic (#2680)

# # 3623

MartinBelthle committed:
refactor(dao): use dao inside the generate thermal timeseries command (#2665)

sylvlecl committed:
perf(raw): avoid double tree parsing when getting file content (#2667)


- we now correctly get the unconsumed URL part and use it
- the PR also fixes some of the many typing issues in inode related code

---------

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

skamril committed:
feat(ui-sts): add occurrences edition in additional constraints form (#2666)

ANT-3538

TheoPascoli committed:
feat(layer): storage isolation (#2677)

Moves layer related methods to a new layer_manager. This manager
implements the DAO pattern.

Adds 3 commands :
- CreateLayer
- RemoveLayer
- UpdateLayerName

There is still something to do regarding layers, it is the updates of
all the areas associated with a given layer. However, this will be done
when the Area has been refactored with the new models.

sylvlecl committed:
refactor(scenario_builder)!: improve scenario builder implementation (#2679)

sylvlecl committed:
feat(scenariobuilder): handle storages additional constraints (#2681)

smailio committed:
fix(ui-districts): update erase areas (#2684)

Co-authored-by: Anis SMAIL <anis.smail_externe@rte-france.com>

sylvlecl commented on antarest/study/dao/db/link.py:
```diff
@@ -0,0 +1,59 @@
+from antarest.study.model import Study
+
+
+class Link(Base):
```
Since the DB objects will anyway be "hidden" behind the DAO layer, I think we'd be better off not using the ORM layer of sqlalchemy, but the core API only.

I had done some testing some time ago that I will try to find, where the plain core API was indeed faster than the ORM layer.

sylvlecl commented on alembic/versions/e316540c7288_add_link_to_bdd.py:
```diff
@@ -0,0 +1,59 @@
+        sa.Column('filter_synthesis', sa.String(), nullable=False),
+        sa.Column('filter_year_by_year', sa.String(), nullable=False),
+        sa.ForeignKeyConstraint(['study_id'], ['study.id'], ondelete='CASCADE'),
+        sa.PrimaryKeyConstraint('study_id', 'area1', 'area2')
```
Not sure for links, but note that for other objects, it will be important to have in the primary key an ID separate from the name. This way, we'll be able to implement renaming easily, for example.

sylvlecl commented on antarest/study/dao/hybrid_dao.py:
```diff
@@ -0,0 +1,76 @@
+from antarest.study.storage.rawstudy.model.filesystem.factory import FileStudy
+
+
+class HybridDao(FileStudyTreeDao):
```
We need to decide the implementation strategy, in particular if we want this kind of hybrid storage or not:
to be honest I think it will make things more complex. But otherwise, we are more or less forced to implement things in a given order, starting with areas.

As discussed:
- one of the first thing we need to do, is also to implement an import feature where we can choose the target DAO (like, simply adding an option in the import API). This way we will be able to use the feature in the app
- we should probably also have a "feature flag" in the configuration to enable or not that feature, until it's completely implemented

sylvlecl reviewed (commented):
Looks promising, quite natural to implement!

MartinBelthle committed:
refactor(comments): introduce dao (#2685)

hdinia committed:
feat(ui-scenario): add sts constraints scenario builder UI (#2687)

skamril committed:
feat(ui-playlist): disable action buttons when no year is selected (#2688)

fill 'Scenario' field with all years by default

hdinia committed:
feat(ui-matrix-filter): enable granular removal of filter values via separate chips (#2682)

sylvlecl committed:
fix(matrix-gc): reduce matrix gc memory usage (#2689)

Try to reduce memory usage of the GC by streaming matrices references
and
keeping only their ID, not the description which can be a pretty large
string for each reference.

---------

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

sylvlecl committed:
fix(scenario): fix scenario builder API regressions (#2691)

Fixes 3 regressions of
https://github.com/AntaresSimulatorTeam/AntaREST/pull/2679 :

- MC year number must start at 0
- fixes a KeyError for studies <= 9.2 which have short term storages
- hydro levels must not be rounded as integers, float values are allowed

Also fixes existing issues in additional constraint management:
- the study config was not correctly updated when creating / removing
storages
- constraint files must be removed when a storage is removed

---------

Signed-off-by: Sylvain Leclerc <sylvain.leclerc@rte-france.com>

MartinBelthle committed:
refactor(playlist)!: introduce dao and new class design (#2683)

MartinBelthle committed:
refactor(user): introduce dao for user commands (#2692)

This PR also allows users to create a file and giving its content. This
way we avoid using the command `UpdateRawFile`

hdinia committed:
refactor(ui-common):  improve `SplitView` prop naming and fix decimal precision in localStorage (#2693)

TheoPascoli committed:
fix(bc): fix a bug when creating new bc (#2699)

The method to get the new id was wrong and would sometimes override
existing bc

hdinia committed:
feat(ui-matrix): add separate enable flags for `Matrix` filters and resize (#2697)

TheoPascoli committed:
Revert "perf(raw): avoid double tree parsing when getting file content (#2667)"

This reverts commit 11d7d3be2cb732f50586336ed39f942337be5f94.

TheoPascoli committed:
Revert "feat(matrix): add arrow format for input matrices inside GET `/raw` endpoint (#2624)"

This reverts commit 7d3bd7b2

smailio committed:
chore(release): v2.24.0 (2025-09-08) (#2701)

Co-authored-by: Anis SMAIL <anis.smail_externe@rte-france.com>

TheoPascoli committed:
Reapply "feat(matrix): add arrow format for input matrices inside GET `/raw` endpoint (#2624)"

This reverts commit 78e02c47946afba5f533d7610282231ca1fa24a3.

TheoPascoli committed:
Reapply "perf(raw): avoid double tree parsing when getting file content (#2667)"

This reverts commit cf0c93615df03f1820f66272a2fa8d264352bd6b.

hdinia committed:
fix(ui-matrix): enable temporal filters for matrices with fixed column names (#2709)

MartinBelthle committed:
feat(sts): add new field `allow-overflow` for v9.3+ studies (#2708)

skamril committed:
feat(ui): use custom dialog instead of window.confirm for unsaved/ongoing submit warning (#2696)

add `useFormCloseConfirm` to show a confirmation dialog when leaving a view with unsaved changes or during submit, even outside react-router

MartinBelthle committed:
build(ts-gen): bump `timeseries-generation` package (#2712)

skamril committed:
feat(ui-launcher): update the dialog (#2711)

- add cluster field
- add cluster metrics
- move fields
- remove adequacy patch fields

hdinia committed:
feat(ui-scenario): filter scenario builder tabs by version (#2698)

skamril committed:
feat(ui-studies): add scope filter and a button to show/hide variants (#2714)

ANT-3022

hdinia committed:
feat(ui-studies): add folder navigation to individual study views (#2710)

ANT-3177

hdinia committed:
feat(ui-matrix): enable temporal filters where missing (#2715)

skamril committed:
fix(ui-studies): minor changes on launch dialog (#2716)

ANT-3022

hdinia committed:
feat(ui-matrix): enable filters on `Inflows` matrix (#2717)

TheoPascoli committed:
chore(import): remove duplicate import of `EnumIgnoreCase`

TheoPascoli committed:
chore: refactor launcher API, improve DTOs, and cleanup unused code

TheoPascoli committed:
chore: refactor launcher API, improve DTOs, and cleanup unused code
